mlflow/models/utils.py

```diff
@@ -50,25 +51,29 @@ class _Example(object):
           encoded strings.
         - numpy types: Numpy types are converted to the corresponding python types or their closest
           equivalent.
+        - csc/csr matric: similar to 2 dims numpy array, csc/csr matric are converted to
+          corresponding python types or their closest equivalent.
     """
 
     def __init__(self, input_example: ModelInputExample):
         def _is_scalar(x):
             return np.isscalar(x) or x is None
 
         def _is_tensor(x):
-            return isinstance(x, np.ndarray) or (
-                isinstance(x, dict) and all([isinstance(ary, np.ndarray) for ary in x.values()])
+            return isinstance(x, (np.ndarray, csr_matrix, csc_matrix)) or (
+                isinstance(x, dict) and all(isinstance(ary, np.ndarray) for ary in x.values())
             )
 
-        def _handle_tensor_input(input_tensor: Union[np.ndarray, dict]):
+        def _handle_tensor_input(input_tensor: Union[np.ndarray, dict, csr_matrix, csc_matrix]):
             if isinstance(input_tensor, dict):
                 result = {}
                 for name in input_tensor.keys():
                     result[name] = input_tensor[name].tolist()
                 return {"inputs": result}
-            else:
+            elif isinstance(input_tensor, np.ndarray):
                 return {"inputs": input_tensor.tolist()}
+            else:
```

Comment:
1. can we do `elif isinstance(input_tesnor,csr_matrix, csc_matrix)`?
2. does this mean we store sparse input as dense vector? or is this stored as array of indices and array of values?

Reply:
I update code to store sparse input as data/indices/indptr vectors.
I also split csc/csr saving code out of the ndarray saving code.